File: app/src/main/java/com/fidisys/android/booksapp/log/TimberThreadDebugTree.java
package com.fidisys.android.booksapp.log;


import timber.log.Timber;

/**
 * Timber base logtree.
 */
public class TimberThreadDebugTree extends Timber.DebugTree {
    @Override
    protected void log(int priority, String tag, String message, Throwable t) {
        String currentThread = Thread.currentThread().getName();
        if (currentThread.trim().isEmpty()) {
            super.log(priority, tag, message, t);
        } else {
            super.log(priority, tag, new StringBuffer(currentThread).append("\n").append(message).toString(), t);
        }
    }
}

File: app/src/main/java/com/fidisys/android/invoiceapp/log/TimberThreadDebugTree.kt
package com.fidisys.android.invoiceapp.log


import timber.log.Timber

/**
 * Timber base logtree.
 */
class TimberThreadDebugTree : Timber.DebugTree() {
    override fun log(priority: Int, tag: String, message: String, t: Throwable?) {
        val currentThread = Thread.currentThread().name
        if (currentThread.trim { it <= ' ' }.isEmpty()) {
            super.log(priority, tag, message, t)
        } else {
            super.log(priority, tag, StringBuffer(currentThread).append("\n").append(message).toString(), t)
        }
    }
}
